### ISSUE TITLE
[Node 18] Global `fetch` API is available by default, but `"fetch" is not defined` is reported

### ISSUE THREAD

**Issue by wesdse001** [eslint/eslint#15941]
### What rule do you want to change?

no-undef

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Other

### Example code

```js
const res = await fetch('https://nodejs.org/api/documentation.json');

if (res.ok) {
  const data = await res.json();
  console.log(data);
}
```

### What does the rule currently do for this code?

The `no-undef` rule throws an error because in earlier version of Node, `fetch` is not defined.

### What will the rule do after it's changed?

For 

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

The `no-undef` rule does not report `"fetch" is not defined` anymore in a node environment.

**Comment by mdjermanovic** [comment]
Hi @wesdse001, thanks for the issue!

This is already supported. Which version of ESLint are you using, and can you share your configuration?

We have updated global variables to include `fetch` in Node environments (https://github.com/eslint/eslintrc/pull/81) as of ESLint v8.16.0.

Here's an [online demo](https://eslint.org/demo#eyJ0ZXh0IjoiY29uc3QgcmVzID0gYXdhaXQgZmV0Y2goJ2h0dHBzOi8vbm9kZWpzLm9yZy9hcGkvZG9jdW1lbnRhdGlvbi5qc29uJyk7XG5cbmlmIChyZXMub2spIHtcbiAgY29uc3QgZGF0YSA9IGF3YWl0IHJlcy5qc29uKCk7XG4gIGNvbnNvbGUubG9nKGRhdGEpO1xufVxuXG5zb21lX3VuZGVmX3ZhcjsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLXVuZGVmIjoyfSwiZW52Ijp7Im5vZGUiOnRydWV9fX0=) with your code example and Node environment enabled, and `no-undef` doesn't report errors on `fetch`.

**Comment by wesdse001** [comment]
Hi @mdjermanovic, thank you so much for the quick reply!

I was using `"eslint": "8.14.0"`.

I just updated to the latest version and everything is good now. I should have check the latest version. Sorry!
